engineio/transport/polling/encoder.go

```diff
@@ -0,0 +1,132 @@
+package polling
+
+import (
+	"encoding/base64"
+	"io"
+	"sync/atomic"
+	"time"
+
+	"github.com/googollee/go-socket.io/engineio/frame"
+)
+
+// encoder encodes frames to the writer.
+type encoder struct {
+	pingTimeout       time.Duration
+	lastPing          time.Time
+	writer            *bufWriter
+	hasFramesChan     chan struct{}
+	closed            chan struct{}
+	hasNonClosedFrame int32
```

Comment:
`hasNonClosedFrame` is it very difficult read to this. i suggest rename this variable to `isClosedFrame`. What do you mean?

Reply:
`hasNonClosedFrame` This means the caller requested a frame to write but didn't close it yet. It's the opposite meaning of `isClosedFrame`.

Reply:
>> It's the opposite meaning of isClosedFrame.

Yeap, sure. you'll change current logic to opposite

Reply:
No. Doing that makes the whole logic complex. When creating a new encoder, `hasNonClosedFrame` is 0 by default, which is natural. When someone calls `Encoder.NextFrame()`, `hasNonClosedFrame` increases.

I know `hasClosedFrame` sounds better, but it's hard to use 0 as default value.